### PR TITLE
fix: Parse MySQL IF statements as ordered compound statements

### LINKED ISSUE #3318: mysql: `IF` statements not supported properly
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

`IF ... THEN ... [ELSEIF ... THEN ...] [ELSE ...] END IF;` is not recognised. Instead, `IF ... THEN ... ;`, `ELSEIF ... THEN ... ;`, and `ELSE ... ;` are each incorrectly recognised as valid independent statements, without any terminating `END IF;`, and the `END` of `END IF;` is treated as the `END` of a `BEGIN ... END;` block, with the `IF` of `END IF;` treated as the label/name/identifier of that block, as in `label: BEGIN ... END label;`.

The incorrect parsing of the `END IF;` causes rules L010 and L014 to be handled incorrectly.

### Expected Behaviour

Parse tree should be something like the following, with grammar element `if_statement = if_clause [elseif_clause] [else_clause] end_if_clause`, and with `if_clause`, `elseif_clause`, and `else_clause` each consisting of their keywords followed by a `statement_block` (if such a thing exists; perhaps the existing `transaction` element can be used?):

```
% echo 'IF TRUE THEN SELECT 1; SELECT 2; ELSEIF FALSE THEN SELECT 3; SELECT 4; ELSE SELECT 5; SELECT 6; END IF;' | sqlfluff parse - --dialect mysql                       
file:
    statement:
        if_statement:
            if_clause:
                keyword:                                      'IF'
                whitespace:                                   ' '
                expression:
                    literal:                                  'TRUE'
                whitespace:                                   ' '
                keyword:                                      'THEN'
                whitespace:                                   ' '
                [META] indent:
                statement_block:
                    statement:
                        select_statement:
                            select_clause:
                                keyword:                      'SELECT'
                                [META] indent:
                                whitespace:                   ' '
                                select_clause_element:
                                    literal:                  '1'
                            [META] dedent:
                    statement_terminator:                     ';'
                    statement:
                        select_statement:
                            select_clause:
                                keyword:                      'SELECT'
                                [META] indent:
                                whitespace:                   ' '
                                select_clause_element:
                                    literal:                  '2'
                            [META] dedent:
                    statement_terminator:                     ';'
            [META] dedent:
            elseif_clause:
                keyword:                                      'ELSEIF'
                whitespace:                                   ' '
                expression:
                    literal:                                  'FALSE'
                whitespace:                                   ' '
                keyword:                                      'THEN'
                whitespace:                                   ' '
                [META] indent:
                statement_block:
                    statement:
                        select_statement:
                            select_clause:
                                keyword:                      'SELECT'
                                [META] indent:
                                whitespace:                   ' '
                                select_clause_element:
                                    literal:                  '3'
                            [META] dedent:
                    statement_terminator:                     ';'
                    statement:
                        select_statement:
                            select_clause:
                                keyword:                      'SELECT'
                                [META] indent:
                                whitespace:                   ' '
                                select_clause_element:
                                    literal:                  '4'
                            [META] dedent:
                    statement_terminator:                     ';'
            [META] dedent:
            else_clause:
                keyword:                                      'ELSE'
                whitespace:                                   ' '
                [META] indent:
                statement_block:
                    statement:
                        select_statement:
                            select_clause:
                                keyword:                      'SELECT'
                                [META] indent:
                                whitespace:                   ' '
                                select_clause_element:
                                    literal:                  '5'
                            [META] dedent:
                    statement_terminator:                     ';'
                    statement:
                        select_statement:
                            select_clause:
                                keyword:                      'SELECT'
                                [META] indent:
                                whitespace:                   ' '
                                select_clause_element:
                                    literal:                  '6'
                            [META] dedent:
                    statement_terminator:                     ';'
            [META] dedent:
            end_if_clause:
                keyword:                                      'END'
                whitespace:                                   ' '
                keyword:                                      'IF'
    statement_terminator:                                     ';'
    newline:                                                  '\n'
```

### Observed Behaviour

`IF` only, with one statement in the block (passes, but parsed incorrectly):

```
% echo 'IF TRUE THEN SELECT 1; END IF;' | sqlfluff parse - --dialect mysql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        if_then_statement:
[L:  1, P:  1]      |            keyword:                                          'IF'
[L:  1, P:  3]      |            whitespace:                                       ' '
[L:  1, P:  4]      |            expression:
[L:  1, P:  4]      |                literal:                                      'TRUE'
[L:  1, P:  8]      |            whitespace:                                       ' '
[L:  1, P:  9]      |            keyword:                                          'THEN'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            statement:
[L:  1, P: 14]      |                select_statement:
[L:  1, P: 14]      |                    select_clause:
[L:  1, P: 14]      |                        keyword:                              'SELECT'
[L:  1, P: 20]      |                        [META] indent:
[L:  1, P: 20]      |                        whitespace:                           ' '
[L:  1, P: 21]      |                        select_clause_element:
[L:  1, P: 21]      |                            literal:                          '1'
[L:  1, P: 22]      |                    [META] dedent:
[L:  1, P: 22]      |    statement_terminator:                                     ';'
[L:  1, P: 23]      |    whitespace:                                               ' '
[L:  1, P: 24]      |    statement:
[L:  1, P: 24]      |        transaction_statement:
[L:  1, P: 24]      |            keyword:                                          'END'
[L:  1, P: 27]      |            whitespace:                                       ' '
[L:  1, P: 28]      |            identifier:                                       'IF'
[L:  1, P: 30]      |    statement_terminator:                                     ';'
[L:  1, P: 31]      |    newline:                                                  '\n'
```

Full `IF ... ELSEIF ... ELSE` (passes, but parsed incorrectly):

```
 % echo 'IF TRUE THEN SELECT 1; SELECT 2; ELSEIF FALSE THEN SELECT 3; SELECT 4; ELSE SELECT 5; SELECT 6; END IF;' | sqlfluff parse - --dialect mysql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        if_then_statement:
[L:  1, P:  1]      |            keyword:                                          'IF'
[L:  1, P:  3]      |            whitespace:                                       ' '
[L:  1, P:  4]      |            expression:
[L:  1, P:  4]      |                literal:                                      'TRUE'
[L:  1, P:  8]      |            whitespace:                                       ' '
[L:  1, P:  9]      |            keyword:                                          'THEN'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            statement:
[L:  1, P: 14]      |                select_statement:
[L:  1, P: 14]      |                    select_clause:
[L:  1, P: 14]      |                        keyword:                              'SELECT'
[L:  1, P: 20]      |                        [META] indent:
[L:  1, P: 20]      |                        whitespace:                           ' '
[L:  1, P: 21]      |                        select_clause_element:
[L:  1, P: 21]      |                            literal:                          '1'
[L:  1, P: 22]      |                    [META] dedent:
[L:  1, P: 22]      |    statement_terminator:                                     ';'
[L:  1, P: 23]      |    whitespace:                                               ' '
[L:  1, P: 24]      |    statement:
[L:  1, P: 24]      |        select_statement:
[L:  1, P: 24]      |            select_clause:
[L:  1, P: 24]      |                keyword:                                      'SELECT'
[L:  1, P: 30]      |                [META] indent:
[L:  1, P: 30]      |                whitespace:                                   ' '
[L:  1, P: 31]      |                select_clause_element:
[L:  1, P: 31]      |                    literal:                                  '2'
[L:  1, P: 32]      |            [META] dedent:
[L:  1, P: 32]      |    statement_terminator:                                     ';'
[L:  1, P: 33]      |    whitespace:                                               ' '
[L:  1, P: 34]      |    statement:
[L:  1, P: 34]      |        if_then_statement:
[L:  1, P: 34]      |            keyword:                                          'ELSEIF'
[L:  1, P: 40]      |            whitespace:                                       ' '
[L:  1, P: 41]      |            expression:
[L:  1, P: 41]      |                literal:                                      'FALSE'
[L:  1, P: 46]      |            whitespace:                                       ' '
[L:  1, P: 47]      |            keyword:                                          'THEN'
[L:  1, P: 51]      |            whitespace:                                       ' '
[L:  1, P: 52]      |            statement:
[L:  1, P: 52]      |                select_statement:
[L:  1, P: 52]      |                    select_clause:
[L:  1, P: 52]      |                        keyword:                              'SELECT'
[L:  1, P: 58]      |                        [META] indent:
[L:  1, P: 58]      |                        whitespace:                           ' '
[L:  1, P: 59]      |                        select_clause_element:
[L:  1, P: 59]      |                            literal:                          '3'
[L:  1, P: 60]      |                    [META] dedent:
[L:  1, P: 60]      |    statement_terminator:                                     ';'
[L:  1, P: 61]      |    whitespace:                                               ' '
[L:  1, P: 62]      |    statement:
[L:  1, P: 62]      |        select_statement:
[L:  1, P: 62]      |            select_clause:
[L:  1, P: 62]      |                keyword:                                      'SELECT'
[L:  1, P: 68]      |                [META] indent:
[L:  1, P: 68]      |                whitespace:                                   ' '
[L:  1, P: 69]      |                select_clause_element:
[L:  1, P: 69]      |                    literal:                                  '4'
[L:  1, P: 70]      |            [META] dedent:
[L:  1, P: 70]      |    statement_terminator:                                     ';'
[L:  1, P: 71]      |    whitespace:                                               ' '
[L:  1, P: 72]      |    statement:
[L:  1, P: 72]      |        if_then_statement:
[L:  1, P: 72]      |            keyword:                                          'ELSE'
[L:  1, P: 76]      |            whitespace:                                       ' '
[L:  1, P: 77]      |            statement:
[L:  1, P: 77]      |                select_statement:
[L:  1, P: 77]      |                    select_clause:
[L:  1, P: 77]      |                        keyword:                              'SELECT'
[L:  1, P: 83]      |                        [META] indent:
[L:  1, P: 83]      |                        whitespace:                           ' '
[L:  1, P: 84]      |                        select_clause_element:
[L:  1, P: 84]      |                            literal:                          '5'
[L:  1, P: 85]      |                    [META] dedent:
[L:  1, P: 85]      |    statement_terminator:                                     ';'
[L:  1, P: 86]      |    whitespace:                                               ' '
[L:  1, P: 87]      |    statement:
[L:  1, P: 87]      |        select_statement:
[L:  1, P: 87]      |            select_clause:
[L:  1, P: 87]      |                keyword:                                      'SELECT'
[L:  1, P: 93]      |                [META] indent:
[L:  1, P: 93]      |                whitespace:                                   ' '
[L:  1, P: 94]      |                select_clause_element:
[L:  1, P: 94]      |                    literal:                                  '6'
[L:  1, P: 95]      |            [META] dedent:
[L:  1, P: 95]      |    statement_terminator:                                     ';'
[L:  1, P: 96]      |    whitespace:                                               ' '
[L:  1, P: 97]      |    statement:
[L:  1, P: 97]      |        transaction_statement:
[L:  1, P: 97]      |            keyword:                                          'END'
[L:  1, P:100]      |            whitespace:                                       ' '
[L:  1, P:101]      |            identifier:                                       'IF'
[L:  1, P:103]      |    statement_terminator:                                     ';'
[L:  1, P:104]      |    newline:                                                  '\n'
```

`ELSEIF ... ELSE` with missing `IF` (passes, but shouldn't):

```
% echo 'ELSEIF FALSE THEN SELECT 3; SELECT 4; ELSE SELECT 5; SELECT 6; END IF;' | sqlfluff parse - --dialect mysql               
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        if_then_statement:
[L:  1, P:  1]      |            keyword:                                          'ELSEIF'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            expression:
[L:  1, P:  8]      |                literal:                                      'FALSE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            keyword:                                          'THEN'
[L:  1, P: 18]      |            whitespace:                                       ' '
[L:  1, P: 19]      |            statement:
[L:  1, P: 19]      |                select_statement:
[L:  1, P: 19]      |                    select_clause:
[L:  1, P: 19]      |                        keyword:                              'SELECT'
[L:  1, P: 25]      |                        [META] indent:
[L:  1, P: 25]      |                        whitespace:                           ' '
[L:  1, P: 26]      |                        select_clause_element:
[L:  1, P: 26]      |                            literal:                          '3'
[L:  1, P: 27]      |                    [META] dedent:
[L:  1, P: 27]      |    statement_terminator:                                     ';'
[L:  1, P: 28]      |    whitespace:                                               ' '
[L:  1, P: 29]      |    statement:
[L:  1, P: 29]      |        select_statement:
[L:  1, P: 29]      |            select_clause:
[L:  1, P: 29]      |                keyword:                                      'SELECT'
[L:  1, P: 35]      |                [META] indent:
[L:  1, P: 35]      |                whitespace:                                   ' '
[L:  1, P: 36]      |                select_clause_element:
[L:  1, P: 36]      |                    literal:                                  '4'
[L:  1, P: 37]      |            [META] dedent:
[L:  1, P: 37]      |    statement_terminator:                                     ';'
[L:  1, P: 38]      |    whitespace:                                               ' '
[L:  1, P: 39]      |    statement:
[L:  1, P: 39]      |        if_then_statement:
[L:  1, P: 39]      |            keyword:                                          'ELSE'
[L:  1, P: 43]      |            whitespace:                                       ' '
[L:  1, P: 44]      |            statement:
[L:  1, P: 44]      |                select_statement:
[L:  1, P: 44]      |                    select_clause:
[L:  1, P: 44]      |                        keyword:                              'SELECT'
[L:  1, P: 50]      |                        [META] indent:
[L:  1, P: 50]      |                        whitespace:                           ' '
[L:  1, P: 51]      |                        select_clause_element:
[L:  1, P: 51]      |                            literal:                          '5'
[L:  1, P: 52]      |                    [META] dedent:
[L:  1, P: 52]      |    statement_terminator:                                     ';'
[L:  1, P: 53]      |    whitespace:                                               ' '
[L:  1, P: 54]      |    statement:
[L:  1, P: 54]      |        select_statement:
[L:  1, P: 54]      |            select_clause:
[L:  1, P: 54]      |                keyword:                                      'SELECT'
[L:  1, P: 60]      |                [META] indent:
[L:  1, P: 60]      |                whitespace:                                   ' '
[L:  1, P: 61]      |                select_clause_element:
[L:  1, P: 61]      |                    literal:                                  '6'
[L:  1, P: 62]      |            [META] dedent:
[L:  1, P: 62]      |    statement_terminator:                                     ';'
[L:  1, P: 63]      |    whitespace:                                               ' '
[L:  1, P: 64]      |    statement:
[L:  1, P: 64]      |        transaction_statement:
[L:  1, P: 64]      |            keyword:                                          'END'
[L:  1, P: 67]      |            whitespace:                                       ' '
[L:  1, P: 68]      |            identifier:                                       'IF'
[L:  1, P: 70]      |    statement_terminator:                                     ';'
[L:  1, P: 71]      |    newline:                                                  '\n'
```

`ELSE` with missing `IF` (passes, but shouldn't):

```
% echo 'ELSE SELECT 5; SELECT 6; END IF;' | sqlfluff parse - --dialect mysql       
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        if_then_statement:
[L:  1, P:  1]      |            keyword:                                          'ELSE'
[L:  1, P:  5]      |            whitespace:                                       ' '
[L:  1, P:  6]      |            statement:
[L:  1, P:  6]      |                select_statement:
[L:  1, P:  6]      |                    select_clause:
[L:  1, P:  6]      |                        keyword:                              'SELECT'
[L:  1, P: 12]      |                        [META] indent:
[L:  1, P: 12]      |                        whitespace:                           ' '
[L:  1, P: 13]      |                        select_clause_element:
[L:  1, P: 13]      |                            literal:                          '5'
[L:  1, P: 14]      |                    [META] dedent:
[L:  1, P: 14]      |    statement_terminator:                                     ';'
[L:  1, P: 15]      |    whitespace:                                               ' '
[L:  1, P: 16]      |    statement:
[L:  1, P: 16]      |        select_statement:
[L:  1, P: 16]      |            select_clause:
[L:  1, P: 16]      |                keyword:                                      'SELECT'
[L:  1, P: 22]      |                [META] indent:
[L:  1, P: 22]      |                whitespace:                                   ' '
[L:  1, P: 23]      |                select_clause_element:
[L:  1, P: 23]      |                    literal:                                  '6'
[L:  1, P: 24]      |            [META] dedent:
[L:  1, P: 24]      |    statement_terminator:                                     ';'
[L:  1, P: 25]      |    whitespace:                                               ' '
[L:  1, P: 26]      |    statement:
[L:  1, P: 26]      |        transaction_statement:
[L:  1, P: 26]      |            keyword:                                          'END'
[L:  1, P: 29]      |            whitespace:                                       ' '
[L:  1, P: 30]      |            identifier:                                       'IF'
[L:  1, P: 32]      |    statement_terminator:                                     ';'
[L:  1, P: 33]      |    newline:                                                  '\n'
```

Incorrect detection of L014 violation:
```
% echo 'IF TRUE THEN SELECT 1 AS x; END IF;' | sqlfluff lint - --dialect mysql 
== [stdin] FAIL                                                                                                                                                                                                                                                                                                                                                                   
L:   1 | P:  33 | L014 | Unquoted identifiers must be consistently lower case.
All Finished 📜 🎉!
```

Incorrect detection of L010 conformance:
```
 % echo 'IF TRUE THEN SELECT 1 AS x; END if;' | sqlfluff lint - --dialect mysql
All Finished 📜 🎉!     
```

Incorrect detection of L010 conformance and L014 violation:
```
echo 'IF TRUE THEN SELECT 1 AS X; END if;' | sqlfluff lint - --dialect mysql
== [stdin] FAIL                                                                                                                                                                                                                                                                                                                                                                   
L:   1 | P:  33 | L014 | Unquoted identifiers must be consistently upper case.
All Finished 📜 🎉!
```

### How to reproduce

See above.

### Dialect

mysql

### Version

```
% sqlfluff --version
sqlfluff, version 0.13.1

% python --version
Python 2.7.16

% python3 --version
Python 3.9.12
```

### Configuration

None/default.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by mvanhorn**
### Brief summary of the change made

Reshape `IfExpressionStatement` in `src/sqlfluff/dialects/dialect_mysql.py` as an ordered sequence: a required `IF ... THEN` branch, repeatable `ELSEIF ... THEN` branches, an optional single `ELSE`, and a required `END IF`.

MySQL procedural `IF` input is currently split into several top-level statements: the opening `IF`, each `ELSEIF` or `ELSE`, and `END IF` all match independently. The current `IfExpressionStatement` uses an unordered `AnyNumberOf`, so it also accepts orphaned branches, missing terminators, duplicate or misordered clauses, and other shapes outside MySQL's stored-program grammar. Recent work means `END IF` is no longer parsed as a labelled `END`, but the checked-in MySQL and inherited MariaDB parse fixtures confirm that the complete construct is still not represented as one nested statement. This plan completes the original issue without changing unrelated procedural constructs such as `WHILE`, `LOOP`, or `REPEAT`.

Fixes #3318

### Are there any other side effects of this change that we should be aware of?

Not applicable to this change.

### Pull Request checklist

- [ ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

AI was used for assistance.

---
## Summary by cubic
Parse MySQL procedural IF as a single ordered compound statement and correctly scope branch bodies. Previously IF/ELSEIF/ELSE/END IF parsed as separate top-level statements and could accept misordered or dangling clauses; now we require IF ... THEN, optional ELSEIF ... THEN branches, optional ELSE, and a closing END IF in order, and nested CASE ELSE no longer terminates the IF branch.

- Introduces an `if_statement_list` segment to hold branch statements with `DelimiterGrammar` and indentation, and excludes `ELSEIF`, `ELSE`, and `END IF` from branch-body matches so nested CASE ELSE remains inside the branch. Updates MySQL and MariaDB fixtures (including nested CASE and multiple ELSEIF cases) and adds tests rejecting invalid IF shapes.

**Migration**
- If you consume the parse tree, update references: branch statements now appear under `if_then_statement.if_statement_list` instead of as sibling top-level statements.

<sup>Written for commit b85439c106f2d876533ac00c4af84ac464fccf16. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sqlfluff/sqlfluff/pull/8380?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -2443,6 +2443,25 @@ class TransactionStatementSegment(BaseSegment):
...
+        Sequence(
+            Ref("StatementSegment"),
+            Ref("DelimiterGrammar"),
+        ),
+        terminators=[
+            "ELSEIF",
+            "ELSE",
+            Sequence("END", "IF"),
+        ],
+        reset_terminators=True,
```
Comment (niklasgruener) [inline src/sqlfluff/dialects/dialect_mysql.py]: Could these terminators also affect parsing inside nested statements?

A `CASE` expression within the `IF` body has its own `ELSE`, which appears to be treated as a terminator for the outer `IF` branch. For example, this leaves an unparsable section:

```sql
IF (x = 0) THEN
    SELECT CASE WHEN a = 1 THEN 1 ELSE 2 END;
END IF;
```

It might be worth keeping these terminators scoped to the top-level branch list and adding this as a regression fixture for MySQL and MariaDB.

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 2]
```diff
@@ -2451,21 +2470,32 @@ class IfExpressionStatement(BaseSegment):
...
-            "THEN",
-            Ref("StatementSegment"),
+    match_grammar = Sequence(
+        "IF",
+        Ref("ExpressionSegment"),
+        "THEN",
+        Indent,
+        Ref("IfStatementListSegment", optional=True),
+        Dedent,
+        AnyNumberOf(
```
Comment (niklasgruener) [inline src/sqlfluff/dialects/dialect_mysql.py:2482]: The repeatable `ELSEIF` path does not seem to be covered directly yet. The existing fixtures contain at most one `ELSEIF`, so a regression that only accepted a single branch could still pass the tests.

A fixture containing two `ELSEIF` branches would lock in the intended behavior. I would also add a negative case without the required `;` before `END IF`, since each statement in the new list requires `DelimiterGrammar`.

**Review by niklasgruener — COMMENTED** [review]
Nice work on bringing the full `IF` construct into one parse tree. I used Context Goblin while reviewing this and left two comments around nested `CASE` parsing and a couple of test edge cases.

**Comment by mvanhorn** [comment]
@niklasgruener good catch on both, fixed in b85439c.

You were right that the terminators leaked into nested statements. Your example did leave an unparsable section. The branch list no longer sets `terminators` / `reset_terminators`; instead the statement ref carries `exclude=OneOf("ELSEIF", "ELSE", Sequence("END", "IF"))`, so the keywords only end a branch when they appear where a statement would start. An `ELSE` belonging to a nested `CASE` is consumed by the `CASE` expression itself and no longer terminates the outer branch. Your snippet is now a fixture in `if_nested.sql` for both MySQL and MariaDB.

On the repeatable path: `if_elseif` now has two consecutive `ELSEIF` branches for both dialects, so a regression that accepted only one branch would fail. The branch list is also `min_times=1` now and the `IfStatementListSegment` refs dropped `optional=True`, which is what makes a missing `;` before `END IF` fail rather than silently parse.

Checked against the full MySQL and MariaDB fixture suite, and your snippet parses with no unparsable sections.

**Review by peterbud — APPROVED** [review]
Thanks @mvanhorn this looks good.
